src/api/api.js

```diff
@@ -19,441 +53,322 @@ export const getMessageConversations = ({
+        resource: `messageConversations/${messageConversationId}/${currentUserId}`,
+        type: 'delete',
+    })
+
+export const markRead = async markedReadConversations =>
+    await engine.mutate({
+        resource: 'messageConversations/read',
+        type: 'create',
+        data: markedReadConversations,
+    })
+
+export const markUnread = async markedUnreadConversations =>
+    await engine.mutate({
+        resource: 'messageConversations/unread',
+        type: 'create',
+        data: markedUnreadConversations,
+    })
+
+export const isInFeedbackRecipientGroup = async currentUser => {
+    const {
+        configuration: { feedbackRecipients },
+    } = await engine.query({
+        configuration: {
+            resource: 'configuration',
+        },
+    })
+    const authorized =
+        currentUser.authorities.includes('ALL') ||
+        currentUser.userGroups.some(({ id }) => id === feedbackRecipients.id)
+    const feedbackRecipientsId = feedbackRecipients.id
+
+    return { authorized, feedbackRecipientsId }
+}
+
+export const searchRecipients = async ({
     searchValue,
     searchOnlyUsers,
     searchOnlyFeedbackRecipients,
     feedbackRecipientsId,
+    dhis2CoreVersion,
 }) => {
-    return getD2Instance()
-        .then(instance => {
-            if (supportsUserLookupEndPoint(instance.system.version.minor)) {
-                // version >= v2.35
-                const url = searchOnlyFeedbackRecipients
-                    ? 'userLookup/feedbackRecipients'
-                    : 'userLookup'
-
-                return instance.Api.getApi().get(`${url}?query=${searchValue}`)
-            } else {
-                // version < 2.35
-                const filters = [`displayName:token:${searchValue}`]
-                if (searchOnlyFeedbackRecipients) {
-                    filters.push(`userGroups.id:eq:${feedbackRecipientsId}`)
-                }
-
-                return instance.Api.getApi().get('users', {
-                    pageSize: MAX_RECIPIENT,
-                    filter: filters,
-                })
-            }
-        })
-        .then(({ users }) =>
-            searchOnlyUsers
-                ? { users, undefined }
-                : searchUserGroups(searchValue).then(({ userGroups }) =>
-                      searchOrganisationUnits(searchValue).then(
-                          ({ organisationUnits }) => ({
-                              users,
-                              userGroups,
-                              organisationUnits,
-                          })
-                      )
-                  )
-        )
-        .catch(error => {
-            throw error
-        })
+    const query = createRecipientSearchQuery({
+        searchValue,
+        searchOnlyUsers,
+        searchOnlyFeedbackRecipients,
+        feedbackRecipientsId,
+        dhis2CoreVersion,
+    })
+
+    const results = await engine.query(query)
+
+    return {
+        users: results.users.users,
```

Comment:
I assume the check here is omitted because we know users will always be present?